app/volume.go

```diff
@@ -354,21 +455,12 @@ func (job *Job) backupAndCleanup() (err error) {
 		return errors.Wrapf(err, "could not get volume %v", volumeName)
 	}
 
-	// Check the size of VolumeHead, if it is empty, skip the recurring backup.
-	// We don't want to overwrite the old backups with new identical backups.
-	// This happens when the volume is detached for a long time and has no new data.
-	volumeHead, err := volumeAPI.ActionSnapshotGet(volume, &longhornclient.SnapshotInput{
-		Name: engineapi.VolumeHeadName,
-	})
-	if err != nil {
-		return errors.Wrapf(err, "could not get volume-head for volume %v", volumeName)
-	}
-	volumeHeadSize, err := strconv.ParseInt(volumeHead.Size, 10, 64)
+	shouldDo, err := job.shouldDoRecurringBackup(volume)
 	if err != nil {
 		return err
 	}
-	if volumeHeadSize == 0 {
-		logrus.Infof("Skipping the recurring backup because volume %v has no new data", volumeName)
+	if !shouldDo {
+		logrus.Infof("Skipping taking backup because volume %v is either empty or doesn't have new data since the last backup", volumeName)
```

Comment:
`volume is empty`? should be `volume head`?

Reply:
When volume head is empty we might still take new backup (e.g. the parent of volume-head is not backuped yet)

Over here, I am referring to the case when volume is just created. The volume is empty (there isn't `lastSnapshot` and the `volume-head` is empty). We cannot compare the  `lastSnapshot`(which is `nil`) with `lastBackup`'s snapshot in this case but we still skip the reccurring backup in this case.